gk/fib.c

```diff
@@ -192,10 +192,19 @@ neigh_get_ether_cache_locked(struct neighbor_hash_table *neigh,
 	 * By calling put_xxx(), the LLS block will call
 	 * gk_arp_and_nd_req_cb(), which, in turn, will call
 	 * clear_ether_cache(), so we can return directly here.
+	 *
+	 * Function get_new_ether_cache_locked() already
+	 * sets @ref_cnt to 1. We need to reset it to 0.
 	 */
+	rte_atomic32_init(&eth_cache->ref_cnt);
```

Comment:
Bug: `eth_cache->ref_cnt` is zero after here, so the entry can be returned by `get_new_ether_cache_locked()` before ` `gk_arp_and_nd_req_cb()` runs. `eth_cache->ref_cnt` must be one to avoid this racing bug.

Reply:
Good point! To deal with the two scenarios described below, we may introduce a separate``bool`` field to indicate whether the entry is being used or not instead of using ``ref_cnt``?

Reply:
> (2) When deleting FIB entries, and there is no reference to the ether cache entries (i.e., ``ref_cnt = 0``). In that case, we will call ``put_xxx()``, which then calls ``gk_arp_and_nd_req_cb()`` to reset the ether cache entry. In this case, the ``ref_cnt`` field is 0, that's why we saw the warning message reported above.

We just need to fix the case above. When `ref_cnt = 1` during a release, just call `put_xxx()` and let it bring `ref_cnt` to zero. Not only are you going to avoid the warning, but you're also going to solve anther racing condition we didn't thing before: while `ref_cnt = 0`, the entry can be reused before `put_xxx()` eventually lead to its release.

You identified the root of the bug correctly, but focused on the warning instead of the cause of it. That warning is that to alert us that something went wrong, it is not the problem.